src/stream/src/executor/source/source_executor.rs

```diff
@@ -564,9 +567,14 @@ impl<S: StateStore> SourceExecutor<S> {
 
                     if let Some(mutation) = barrier.mutation.as_deref() {
                         match mutation {
-                            // XXX: Is it possible that the stream is self_paused, and we have pause mutation now? In this case, it will panic.
```

Comment:
pause_stream and resume_stream have been modified to warn instead of panic when seeing unexpected pause/resume.